etc/notes/CHANGES_6.0.0.md

```diff
@@ -2,23 +2,339 @@
+
+## ⚙️ Option parsing improvements
+
+### `useNewUrlParser` and `useUnifiedTopology` emit deprecation warnings
+
+These options were removed in 4.0.0 but continued to be parsed and silently left unused. We have now added a deprecation warning through Node.js' [warning system](https://nodejs.org/api/process.html#event-warning) and will fully remove these options in the _next_ major release.
+
+### Boolean options only accept 'true' or 'false' in connection strings
+
+Prior to this change, we accepted the values `'1', 'y', 'yes', 't'` as synonyms for `true` and `'-1', '0', 'f', 'n', 'no'` as synonyms for `false`. These have now been removed in an effort to make working with connection string options simpler.
+
+```ts
+// Incorrect
+const client = new MongoClient('mongodb://localhost:27017?tls=1'); // throws MongoParseError
+
+// Correct
+const client = new MongoClient('mongodb://localhost:27017?tls=true');
+```
+
+### Repeated options in connection string are no longer allowed
+
+In order to avoid accidental misconfiguration the driver will no longer prioritize the first instance of an option provided on the URI. Instead repeated options that are not permitted to be repeated will throw an error. This is to that connection strings that contain options like `tls=true&tls=false` are no longer ambiguous.
+
+### TLS certificate authority and certificate-key files are now read asynchronously
+
+In order to align with Node.js best practices of keeping I/O async, we have updated the `MongoClient` to store the file names provided to the existing `tlsCAFile` and `tlsCertificateKeyFile` options and only read these files the first time it connects. Prior to this change, the files were read synchronously on `MongoClient` construction.
+
+> [!NOTE]
+> This has no effect on driver functionality when TLS configuration files are properly specified. However, if there are any issues with the TLS configuration files (invalid file name), the error is now thrown when the `MongoClient` is connected instead of at construction time.
+
+```ts
+const client = new MongoClient(CONNECTION_STRING, {
+  tls: true,
+  tlsCAFile: 'caFileName',
+  tlsCertificateKeyFile: 'certKeyFile'
+}); // Files are not read here, but file names are stored on the MongoClient
+
+await client.connect(); // Files are now read and their contents stored
+await client.close();
+
+await client.connect(); // Since the file contents have already been cached, the files will not be read again.
+```
+
+Take a look at our [TLS documentation](https://www.mongodb.com/docs/drivers/node/current/fundamentals/connection/tls/) for more information on the `tlsCAFile` and `tlsCertificateKeyFile` options.
```

Comment:
```suggestion
### TLS certificate authority and certificate-key files are now read asynchronously

In order to align with Node.js best practices of keeping I/O async, we have updated the `MongoClient` to store the file names provided to the existing `tlsCAFile` and `tlsCertificateKeyFile` options, as well as the new `tlsCRLFile` option, and only read these files the first time it connects. Prior to this change, the files were read synchronously on `MongoClient` construction.

> [!NOTE]
> This has no effect on driver functionality when TLS configuration files are properly specified. However, if there are any issues with the TLS configuration files (invalid file name), the error is now thrown when the `MongoClient` is connected instead of at construction time.

```ts
const client = new MongoClient(CONNECTION_STRING, {
  tls: true,
  tlsCAFile: 'caFileName',
  tlsCertificateKeyFile: 'certKeyFile',
  tlsCRLFile: 'crlPemFile'
}); // Files are not read here, but file names are stored on the MongoClient

await client.connect(); // Files are now read and their contents stored
await client.close();

await client.connect(); // Since the file contents have already been cached, the files will not be read again.
```

Take a look at our [TLS documentation](https://www.mongodb.com/docs/drivers/node/current/fundamentals/connection/tls/) for more information on the `tlsCAFile`, `tlsCertificateKeyFile` and `tlsCRLFile` options.
```

Reply:
Oh Github had a problem with double closing "```". Hope this is clear.

Reply:
Updated manually